// hass-ts/dist/entities/calendar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calendar = void 0;
const _core_1 = require("../core");
class Calendar {
    constructor(id, client) {
        this.id = id;
        this.entity = new _core_1.Entity(this.id, client);
    }
    isEventCurrentlyHappening() {
        return this.entity.state.state === "on";
    }
    get currentEvent() {
        if (this.entity.state.entity_id !== this.id) {
            throw new Error("State id did not match id. Something has gone wrong");
        }
        return this.parseAttributes(this.entity.state.attributes);
    }
    parseAttributes(attributes) {
        return {
            ...attributes,
            start: new Date(attributes.start_time),
            end: new Date(attributes.end_time),
        };
    }
    onStartEvent(callback) {
        this.entity.onStateChanged((oldState, newState) => {
            if (oldState.state === "off" && newState.state === "on") {
                if (newState.entity_id === this.id) {
                    callback(this.parseAttributes(newState.attributes));
                }
            }
        });
    }
    onFinishEvent(callback) {
        this.entity.onStateChanged((oldState, newState) => {
            if (oldState.state === "on" && newState.state === "off") {
                callback();
            }
        });
    }
}
exports.Calendar = Calendar;
