// hass-ts/dist/entities/climate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Climate = void 0;
const _core_1 = require("../core");
const service_command_1 = require("../core/service-command");
class Climate {
    constructor(id, client) {
        this.id = id;
        this.entity = new _core_1.Entity(this.id, client);
        this.setAuxHeatCommand = new service_command_1.ServiceCommand("climate.set_aux_heat", client);
        this.setHvacModeCommand = new service_command_1.ServiceCommand("climate.set_hvac_mode", client);
        this.setPresetModeCommand = new service_command_1.ServiceCommand("climate.set_preset_mode", client);
        this.setTemperatureCommand = new service_command_1.ServiceCommand("climate.set_temperature", client);
    }
    async setTemperature(args) {
        this.setTemperatureCommand.call({ ...args, entity_id: this.id });
    }
    async setAuxHeat(auxHeat) {
        this.setAuxHeatCommand.call({
            entity_id: this.id,
            aux_heat: auxHeat,
        });
    }
    async setHvacMode(mode) {
        this.setHvacModeCommand.call({
            entity_id: this.id,
            hvac_mode: mode,
        });
    }
    async setPresetMode(mode) {
        this.setPresetModeCommand.call({
            entity_id: this.id,
            preset_mode: mode,
        });
    }
    onStateChange(callback) {
        this.entity.onStateChanged((oldState, newState) => {
            if (oldState.entity_id === this.id && newState.entity_id === this.id) {
                callback(oldState, newState);
            }
        });
    }
}
exports.Climate = Climate;
