// hass-ts/dist/core/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
const constants_1 = require("./constants");
const home_assistant_api_1 = require("./home-assistant-api");
const _utils_1 = require("../utils");
const get_config_1 = require("./get-config");
class Client {
    constructor(hassConfig, logger) {
        this.hassConfig = hassConfig;
        this.logger = logger;
        this.states = new Map();
        this.timers = [];
        this.stateLoadCallbacks = new Map();
        this.stateChangedCallbacks = new Map();
        this.hassApi = new home_assistant_api_1.HomeAssistantApi(this.hassConfig);
        const loadStateTimers = setInterval(async () => {
            await this.loadStates();
        }, constants_1.FIVE_MINUTES);
        this.timers.push(loadStateTimers);
    }
    /**
     * Create a new Hass client instance.
     */
    static async start(logger) {
        const config = (0, get_config_1.getConfig)();
        const client = new Client(config, logger);
        await client.init();
        return client;
    }
    onStateLoaded(entityId, callback) {
        const storedCallbacks = this.stateLoadCallbacks.get(entityId);
        this.stateLoadCallbacks.set(entityId, [
            ...(storedCallbacks ?? []),
            callback,
        ]);
    }
    stateChangedListener(event) {
        const callbacks = this.stateChangedCallbacks.get(event.data.entity_id);
        callbacks?.forEach((callback) => callback(event.data.old_state, event.data.new_state));
    }
    onStateChanged(entityId, callback) {
        const storedCallbacks = this.stateLoadCallbacks.get(entityId);
        this.stateChangedCallbacks.set(entityId, [
            ...(storedCallbacks ?? []),
            callback,
        ]);
    }
    async callService(domain, service, fields) {
        await this.hassApi.get.callService(domain, service, fields);
    }
    removeStateChangedCallback(entityId, callback) {
        const storedCallbacks = this.stateChangedCallbacks.get(entityId);
        if (storedCallbacks) {
            const index = storedCallbacks.findIndex((needle) => needle === callback);
            this.stateChangedCallbacks.set(entityId, (0, _utils_1.removeItemAtIndex)(storedCallbacks, index));
        }
    }
    removeOnStateLoadedCallback(entityId, callback) {
        const storedCallbacks = this.stateLoadCallbacks.get(entityId);
        if (storedCallbacks) {
            const index = storedCallbacks.findIndex((needle) => needle === callback);
            this.stateLoadCallbacks.set(entityId, (0, _utils_1.removeItemAtIndex)(storedCallbacks, index));
        }
    }
    parseState(state) {
        return {
            ...state,
            last_changed: new Date(state.last_changed),
            last_updated: new Date(state.last_updated),
        };
    }
    async loadStates() {
        this.logger.debug(`Loading states`);
        const states = await this.hassApi.get.getStates();
        const stateMap = new Map();
        states
            .map((state) => this.parseState(state))
            .forEach((state) => {
            stateMap.set(state.entity_id, state);
            const callbacks = this.stateLoadCallbacks.get(state.entity_id);
            callbacks?.forEach((callback) => callback(state));
        });
        this.logger.debug(`Finished loading states`);
        this.states = stateMap;
    }
    close() {
        this.hassApi.close();
        this.timers.forEach((timer) => clearInterval(timer));
        this.logger.info(`Hass client closed`);
    }
    async init() {
        await this.hassApi.init();
        this.hassApi.get.on("state_changed", this.stateChangedListener.bind(this));
        await this.loadStates();
        this.logger.info("Hass client initialised");
    }
    cachedStates() {
        return this.states;
    }
}
exports.Client = Client;
