// hass-ts/dist/core/entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Entity = void 0;
class Entity {
    constructor(id, client) {
        this.id = id;
        this.client = client;
        this.stateLoadedListener = this.onStateLoaded.bind(this);
        this.client.onStateLoaded(this.id, this.stateLoadedListener);
        const state = this.client.cachedStates().get(this.id);
        if (!state) {
            throw new Error("State could not be loaded. That means either the entity id is wrong, or the client has not been initialised");
        }
        this._state = state;
        this.client.onStateChanged(id, (_, newState) => {
            this._state = newState;
        });
    }
    onStateLoaded(state) {
        this._state = state;
        this.client.removeOnStateLoadedCallback(this.id, this.stateLoadedListener);
    }
    onStateChanged(callback) {
        this.client.onStateChanged(this.id, callback);
    }
    get state() {
        return this._state;
    }
    stateListener() { }
}
exports.Entity = Entity;
