// hass-ts/dist/core/home-assistant-api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HomeAssistantApi = void 0;
const homeassistant_ws_1 = __importDefault(require("homeassistant-ws"));
const isomorphic_ws_1 = __importDefault(require("isomorphic-ws"));
class HomeAssistantApi {
    constructor(hassConfig) {
        this.hassConfig = hassConfig;
    }
    get get() {
        if (!this.hassApi) {
            throw new Error("Hass API has not been initialised. Please call .init()");
        }
        return this.hassApi;
    }
    close() {
        this.hassApi = undefined;
    }
    async init() {
        if (!this.hassApi) {
            const ws = !this.hassConfig.port
                ? {
                    ws: (opts) => {
                        return new isomorphic_ws_1.default(`${opts.protocol}://${opts.host}${opts.path}`);
                    },
                }
                : {};
            this.hassApi = await (0, homeassistant_ws_1.default)({
                host: this.hassConfig.host,
                token: this.hassConfig.token,
                port: this.hassConfig.port,
                path: this.hassConfig.path,
                ...ws,
            });
        }
    }
}
exports.HomeAssistantApi = HomeAssistantApi;
